unitTests/ui/ResponsiveComponents/ResponsiveComponentsManagerTest.ts

```diff
@@ -36,19 +46,27 @@ export function ResponsiveComponentsManagerTest() {
 
     it('calls handle resize event when resize listener is called', () => {
       root.width = () => 400;
-      responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
-      responsiveComponentsManager.resizeListener();
-      jasmine.clock().tick(250);
+      setupHandleResizeEventTest();
       expect(handleResizeEvent).toHaveBeenCalled();
     });
 
     it('does not calls handle resize event when resize listener is called and width is zero', () => {
       root.width = () => 0;
-      responsiveComponentsManager.register(responsiveComponent, root, 'id', component, {});
-      jasmine.clock().tick(250);
+      setupHandleResizeEventTest();
+      expect(handleResizeEvent).not.toHaveBeenCalled();
+    });
 
-      responsiveComponentsManager.resizeListener();
+    ['small', 'medium', 'large'].forEach((responsiveMode: ValidResponsiveMode) => {
```

Comment:
For consistency's sake, I would create one test, and loop inside the test itself, versus generating the test function like this.

What you did is okay, it's just not used anywhere else in other test files, so it looks a bit out of place.

Reply:
It's just to have more meaning full test failing message.
With jasmine, you can't can add message detail on the failing expectation.